src/Aspire.Hosting.Azure.Storage/AzureStorageResource.cs

```diff
@@ -83,4 +86,16 @@ void IResourceWithAzureFunctionsConfig.ApplyAzureFunctionsConfiguration(IDiction
             target[$"{QueuesConnectionKeyPrefix}__{connectionName}__ServiceUri"] = QueueEndpoint;
         }
     }
+
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name="name"></param>
+    /// <returns></returns>
+    public override ProvisionableResource GetExistingResource(BicepValue<string> name)
+    {
+        var account = StorageAccount.FromExisting(Infrastructure.NormalizeBicepIdentifier(Name));
+        account.Name = name;
```

Comment:
This the strangest part of this change. We need the ability to import the underlying resource use for the role assignment given a name. Look at [this](https://github.com/dotnet/aspire/pull/6636/files#diff-5c542bf0932589f40e044926646e54c8fed701df0595fcdf18025abbe80aaea3R11) to understand more.

This method is called from [here](https://github.com/dotnet/aspire/pull/6636/files#diff-40fca6106cdf3968888c9d2bf55bb47a99dd1087a005c291248d2783dcd5dc07R330)